scripts/perf/scheduler_ops_metrics.py

```diff
@@ -0,0 +1,172 @@
+# -*- coding: utf-8 -*-
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from datetime import datetime
+import logging
+import pandas as pd
+import sys
+
+from airflow import configuration, settings
+from airflow.jobs import SchedulerJob
+from airflow.models import DagModel, DagRun, TaskInstance
+from airflow.utils.state import State
+
+DAG_IDS = ['perf_dag_1', 'perf_dag_2']
+MAX_RUNTIME_SECS = 600
+
+
+class SchedulerMetricsJob(SchedulerJob):
+    """
+    This class extends SchedulerJob to instrument the execution performance of
+    task instances contained in each DAG. We want to know if any DAG
+    is starved of resources, and this will be reflected in the stats printed
+    out at the end of the test run. The following metrics will be instrumented
+    for each task instance (dag_id, task_id, execution_date) tuple:
+
+    1. Queuing delay - time taken from starting the executor to the task
+       instance to be added to the executor queue.
+    2. Start delay - time taken from starting the executor to the task instance
+       to start execution.
+    3. Land time - time taken from starting the executor to task instance
+       completion.
+    4. Duration - time taken for executing the task instance.
+
+    The DAGs implement bash operators that call the system wait command. This
+    is representative of typical operators run on Airflow - queries that are
+    run on remote systems and spend the majority of their time on I/O wait.
+
+    To Run:
+        $ python scripts/perf/scheduler_ops_metrics.py
+    """
+    __mapper_args__ = {
+        'polymorphic_identity': 'SchedulerMetricsJob'
+    }
+
+    def print_stats(self):
```

Comment:
Would be good to print a warning or error for each TI that was not in the success state.